notification-portlet-webapp/src/main/java/org/jasig/portlet/notice/service/jpa/JpaNotificationDao.java

```diff
@@ -84,6 +84,22 @@ public JpaEntry getFullEntry(long entryId) {
         return query.getResultList();
     }
 
+    @Override
+    @Transactional(readOnly = true)
+    public List<JpaEntry> getNotificationsBySourceAndCustomAttribute(String source, String attributeName, String attributeValue) {
+        final String jpql = "SELECT e FROM JpaEntry e "
+                + "WHERE e.source = :source "
+                + "AND EXISTS ("
+                    + "SELECT a FROM JpaAttribute a "
```

Comment:
:art: maybe use explicit alias `FROM JpaAttribute AS a` instead of shorthand `FROM JpaAttribute a`?